gef.py

```diff
@@ -5207,6 +5205,199 @@ def do_invoke(self, args: List) -> None:
+                    return values(value)
+                try:
+                    for val, desc in values:
+                        if value == val: return desc
+                        if val is None: default = desc
+                except:
+                    err(f"Error while trying to obtain values from _values_[\"{name}\"]")
+            return default
+
+    class Module:
+        def __init__(self, path: pathlib.Path) -> None:
+            self.path = path
+            self.raw = self.__load()
+            self.structures : List[ExternalStructureManager.Structure] = []
+
+            for entry in self:
+                self.structures.append(ExternalStructureManager.Structure(self.path, entry))
+            return
+
+        def __load(self) -> ModuleType:
+            """Load a custom module, and return it"""
+            fpath = self.path
+            spec = importlib.util.spec_from_file_location(fpath.stem, fpath)
+            module = importlib.util.module_from_spec(spec)
+            sys.modules[fpath.stem] = module
+            spec.loader.exec_module(module)
+            return module
+
+        def __str__(self) -> str:
+            return self.path.stem
+
+        def __iter__(self) -> Generator[str, None, None]:
+            _invalid = {"BigEndianStructure", "LittleEndianStructure", "Structure"}
+            _structs = {x for x in dir(self.raw) \
+                             if inspect.isclass(getattr(self.raw, x)) \
+                             and issubclass(getattr(self.raw, x), ctypes.Structure)}
+            for entry in (_structs - _invalid):
+                yield entry
+            return
+
+        def __contains__(self, structure_name: str) -> bool:
+            for structure in self.structures:
+                if structure.name == structure_name:
+                    return True
+            return False
+
+        def __getitem__(self, structure_name: str) -> "ExternalStructureManager.Structure":
+            for structure in self.structures:
+                print(self.path, structure.name)
+                if structure.name == structure_name:
+                    return structure
+            raise KeyError(f"Structure {structure_name} not found")
+
+    class Modules:
+        def __init__(self, path: pathlib.Path) -> None:
+            self.modules: List[ExternalStructureManager.Module] = []
+            self.root: pathlib.Path = path
+
+            for entry in self.root.iterdir():
+                if not entry.is_file(): continue
+                if entry.suffix != ".py": continue
+                if entry.name == "__init__.py": continue
+                self.modules.append(ExternalStructureManager.Module(entry))
+            return
+
+        def __contains__(self, structure_name: str) -> bool:
+            """Return True if the structure name is found in any of the modules"""
+            for module in self.modules:
+                if structure_name in [s.name for s in module.structures]:
```

Comment:
you're basically re-implementing `module.__contains__`. You should just do:

```python
for module in self.modules:
    if structure_name in module:
        return True
```

that way the `in` will call the `__contains__` you wrote above.